tools/istio-iptables/main.go

```diff
@@ -0,0 +1,516 @@
+	flagSet.StringVar(&proxyUID, "u", proxyUID,
+		"Specify the UID of the user for which the redirection is not applied. Typically, this is the UID of the proxy container")
+	flagSet.StringVar(&proxyGID, "g", proxyGID,
+		"Specify the GID of the user for which the redirection is not applied. (same default value as -u param)")
+	flagSet.StringVar(&inboundInterceptionMode, "m", inboundInterceptionMode,
+		"The mode used to redirect inbound connections to Envoy, either \"REDIRECT\" or \"TPROXY\"")
+	flagSet.StringVar(&inboundPortsInclude, "b", inboundPortsInclude,
+		"Comma separated list of inbound ports for which traffic is to be redirected to Envoy (optional). "+
+			"The wildcard character \"*\" can be used to configure redirection for all ports. An empty list will disable")
+	flagSet.StringVar(&inboundPortsExclude, "d", inboundPortsExclude,
+		"Comma separated list of inbound ports to be excluded from redirection to Envoy (optional). "+
+			"Only applies  when all inbound traffic (i.e. \"*\") is being redirected (default to $ISTIO_LOCAL_EXCLUDE_PORTS)")
+	flagSet.StringVar(&outboundIPRangesInclude, "i", outboundIPRangesInclude,
+		"Comma separated list of IP ranges in CIDR form to redirect to envoy (optional). "+
+			"The wildcard character \"*\" can be used to redirect all outbound traffic. An empty list will disable all outbound")
+	flagSet.StringVar(&outboundIPRangesExclude, "x", outboundIPRangesExclude,
+		"Comma separated list of IP ranges in CIDR form to be excluded from redirection. "+
+			"Only applies when all  outbound traffic (i.e. \"*\") is being redirected (default to $ISTIO_SERVICE_EXCLUDE_CIDR)")
+	flagSet.StringVar(&outboundPortsExclude, "o", outboundPortsExclude,
+		"Comma separated list of outbound ports to be excluded from redirection to Envoy (optional")
+	flagSet.StringVar(&kubevirtInterfaces, "k", kubevirtInterfaces,
+		"Comma separated list of virtual interfaces whose inbound traffic (from VM) will be treated as outbound (optional)")
+
+	var dryRun bool
+	flagSet.BoolVar(&dryRun, "dryRun", false, "Do not call any external dependencies like iptables")
+	err := flagSet.Parse(args)
+	if err != nil {
+		return
+	}
+
+	var ext dep.Dependencies
+	if dryRun {
+		ext = &dep.StdoutStubDependencies{}
+	} else {
+		ext = &dep.RealDependencies{}
+	}
+
+	defer func() {
+		ext.RunOrFail(dep.IPTABLESSAVE)
+		ext.RunOrFail(dep.IP6TABLESSAVE)
+	}()
+
+	// TODO: more flexibility - maybe a whitelist of users to be captured for output instead of a blacklist.
+	if proxyUID == "" {
+		usr, err := ext.LookupUser()
+		var userID string
+		// Default to the UID of ENVOY_USER and root
+		if err != nil {
+			userID = "1337"
+		} else {
+			userID = usr.Uid
+		}
+		// If ENVOY_UID is not explicitly defined (as it would be in k8s env), we add root to the list,
+		// for ca agent.
+		proxyUID = userID + ",0"
+	}
+
+	// for TPROXY as its uid and gid are same
+	if proxyGID == "" {
+		proxyGID = proxyUID
+	}
+
+	podIP, err := ext.GetLocalIP()
```

Comment:
comment the purpose here

Reply:
See https://github.com/istio/istio/pull/14690#issuecomment-501282427